src/name/bizna/ocarmsim/BasicDebugger.java

```diff
@@ -0,0 +1,248 @@
+package name.bizna.ocarmsim;
+
+import java.awt.Component;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import li.cil.oc.api.machine.ExecutionResult;
+import name.bizna.jarm.AlignmentException;
+import name.bizna.jarm.BusErrorException;
+import name.bizna.jarm.CPU;
+import name.bizna.jarm.Debugger;
+import name.bizna.jarm.UndefinedException;
+import name.bizna.jarm.UnimplementedInstructionException;
+import name.bizna.jarm.VirtualMemorySpace;
+
+/**
+ *
+ * @author Jean-Rémy Buchs <jrb0001@692b8c32.de>
+ */
+public abstract class BasicDebugger extends Debugger implements Runnable {
+
+	private State state;
+	private long sleep;
+	private String reason;
+	private boolean singleStep;
+	private final Set<Breakpoint> breakpoints = new HashSet<>();
+	private final Set<Breakpoint> readWatchpoints = new HashSet<>();
+	private final Set<Breakpoint> writeWatchpoints = new HashSet<>();
+
+	public synchronized void reset() {
+		// Reset CPU.
+		cpu.reset(false, true, true);
+
+		// Reset components.
+		((FakeMachine) ((CP3) cpu.getCoprocessor(3)).getMachine()).getComponents().forEach((c) -> {
+			c.reset();
+		});
```

Comment:
I didn't see any other Java 8 classes or features in my review, but I don't have Java 7 myself...